index.js

```diff
@@ -68,22 +68,31 @@ class SimpleKeyring extends EventEmitter {
   }
 
   // For personal_sign, we need to prefix the message:
-  signPersonalMessage (withAccount, msgHex) {
-    const wallet = this._getWalletForAccount(withAccount)
+  signPersonalMessage (withAccount, msgHex, opts) {
+    const wallet = this._getWalletForAccount(withAccount, opts)
     const privKey = ethUtil.stripHexPrefix(wallet.getPrivateKey())
     const privKeyBuffer = new Buffer(privKey, 'hex')
     const sig = sigUtil.personalSign(privKeyBuffer, { data: msgHex })
     return Promise.resolve(sig)
   }
 
   // personal_signTypedData, signs data along with the schema
-  signTypedData (withAccount, typedData) {
-    const wallet = this._getWalletForAccount(withAccount)
+  signTypedData (withAccount, typedData, opts) {
+    const wallet = this._getWalletForAccount(withAccount, opts)
     const privKey = ethUtil.toBuffer(wallet.getPrivateKey())
     const sig = sigUtil.signTypedData(privKey, { data: typedData })
     return Promise.resolve(sig)
   }
 
+  // returns an address specific to an app
+  getAppKeyAddress (address, origin) {
```

Comment:
This method is a mix of sync and async - it returns the result as a Promise, but any errors encountered would be thrown synchronously rather than returned as a `Promise.reject`.

I'd suggest either making this synchronous, or making it `async`. Or if you want it to be `async` but don't want to introduce async/await into this library, you could use `try...catch` to ensure a `Promise.reject` is returned.

Reply:
That's a good catch. Making it `async` seems like it fixes the mistake, as the method is meant to be promise-returning.

Reply:
Done in 71b4160

Reply:
I think the `async` keyword would've saved _a little_ code here, but nbd.